mosaic-runtime/src/commonTest/kotlin/com/jakewharton/mosaic/ui/RowColumnTest.kt

```diff
@@ -0,0 +1,2399 @@
+	}
+
+	@Test fun testRow_withChildrenWithWeightNonFilling() {
+		val width = 50
+		val height = 80
+
+		val rootNode = mosaicNodesWithMeasureAndPlace {
+			Container(
+				modifier = Modifier.sizeIn(maxWidth = 300, maxHeight = 300),
+				alignment = Alignment.TopStart,
+			) {
+				Row {
+					Container(modifier = Modifier.weight(1.0f, fill = false), width = width, height = height)
+					Container(
+						modifier = Modifier.weight(2.0f, fill = false),
+						width = width,
+						height = height * 2,
+					)
+				}
+			}
+		}
+
+		val firstChildContainerNode = rootNode.children[0].children[0].children[0]
+		val secondChildContainerNode = rootNode.children[0].children[0].children[1]
+
+		assertThat(firstChildContainerNode.size).isEqualTo(IntSize(width, height))
+		assertThat(firstChildContainerNode.position).isEqualTo(IntOffset.Zero)
+		assertThat(secondChildContainerNode.size).isEqualTo(IntSize(width, height * 2))
+		assertThat(secondChildContainerNode.position).isEqualTo(IntOffset(width, 0))
+	}
+
+	@Test fun testRow_withChildrenWithMaxValueWeight() {
+		val width = 50
+		val height = 80
+
+		val rootNode = mosaicNodesWithMeasureAndPlace {
+			Container(alignment = Alignment.TopStart) {
+				Row {
+					Container(modifier = Modifier.weight(Float.MAX_VALUE), width = width, height = height)
+					Container(modifier = Modifier.weight(1.0f), width = width, height = height)
+				}
+			}
+		}
+
+		val rootWidth = rootNode.width
+
+		val firstChildContainerNode = rootNode.children[0].children[0].children[0]
+		val secondChildContainerNode = rootNode.children[0].children[0].children[1]
+
+		assertThat(firstChildContainerNode.size).isEqualTo(IntSize(rootWidth, height))
+		assertThat(firstChildContainerNode.position).isEqualTo(IntOffset.Zero)
+		assertThat(secondChildContainerNode.size).isEqualTo(IntSize(0, height))
+		assertThat(secondChildContainerNode.position).isEqualTo(IntOffset(rootWidth, 0))
+	}
+
+	@Test fun testRow_withChildrenWithPositiveInfinityWeight() {
+		val width = 50
+		val height = 80
+
+		val rootNode = mosaicNodesWithMeasureAndPlace {
+			Container(alignment = Alignment.TopStart) {
+				Row {
+					Container(
+						modifier = Modifier.weight(Float.POSITIVE_INFINITY),
+						width = width,
+						height = height,
+					)
+					Container(modifier = Modifier.weight(1.0f), width = width, height = height)
+				}
+			}
+		}
+
+		val rootWidth = rootNode.width
+
+		val firstChildContainerNode = rootNode.children[0].children[0].children[0]
+		val secondChildContainerNode = rootNode.children[0].children[0].children[1]
+
+		assertThat(firstChildContainerNode.size).isEqualTo(IntSize(rootWidth, height))
+		assertThat(firstChildContainerNode.position).isEqualTo(IntOffset.Zero)
+		assertThat(secondChildContainerNode.size).isEqualTo(IntSize(0, height))
+		assertThat(secondChildContainerNode.position).isEqualTo(IntOffset(rootWidth, 0))
+	}
+
+	@Test fun testRow_invalidWeight() {
+		with(RowScopeInstance) {
+			assertFails {
```

Comment:
Would like to see at minimum asserting on the exception type here, if not also the message.